fifo.c

```diff
@@ -2,82 +2,317 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <time.h>
 #include <sys/time.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdbool.h>
 
+#define SECONDS_PER_MINUTE 60
+#define SIZE_OF_USED_BUFFER 24    /* we put a sequence number and a timestamp at the beginning
+                                  of the buffer */
+#define ARRAY_BOOST 20000         /* The size of the array initially and how much to increase it's
+                                  size as it grows */  
+
+/* Author:   Matt Currier
+ * 
+ * This code base is from :
+ * https://openbenchmarking.org/test/pts/ipc-benchmark ->
+ * https://github.com/detailyang/ipc_benchmark
+ *
+ * MIT License
+ * 
+ * Copyright (c) 2017 detailyang
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * 
+ * To find out what cpus share a core (e.g. L1, L2 caches), on intel, 
+ * grep . /sys/devices/system/cpu/cpu*\/topology/thread_siblings_list
+ * Stay away from a core-pair that has cpu0 in it. 
+ *
+ */
```

Comment:
We have another open PR to update the license from MIT to Apache 2.0, as discussed. This license statement should not be added here since it contradicts the other licensing effort.